src/main/java/org/testng/IDataProviderMethod.java

```diff
@@ -0,0 +1,41 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+/**
+ * Represents the attributes of a {@link org.testng.annotations.DataProvider} annotated method.
+ */
+public interface IDataProviderMethod {
+    /**
+     * @return - The instance to which the data provider belongs to. <code>null</code> if the data provider
+     * is a static one.
+     */
+    Object getInstance();
+
+    /**
+     * @return - A {@link Method} object that represents the actual
+     * {@literal @}{@link org.testng.annotations.DataProvider} method.
+     */
+    Method getMethod();
+
+    /**
+     * The name of this DataProvider.
+     */
+    String getName();
+
+    /**
+     * Whether this data provider should be run in parallel.
+     */
+    boolean isParallel();
+
+    /**
+     * Which indices to run from this data provider, default: all.
+     */
+    List<Integer> getIndices();
+
+    /**
+     * @return the current {@link ITestContext}.
+     */
+    ITestContext getTestContext();
```

Comment:
I added it to cater to the following needs : 

* What if the user needed access to the parameters section of a current `<test>` ? 
* What if the user wanted to share some attributes with the listeners via the context object ?
* What if the user wanted to some specific logic based on the knowledge of knowing what was the current <test> being executed ?
* All our listeners are providing access to the `ITestContext` object in one way or the other, except for this listener. 

Please let me know if that sounds fair enough.